Improves import and installation documentation (#36)
wwt/SwiftCurrent: wiki/Installation.md

# Swift Package Manager

If you want more information about installing Swift packages through Xcode, [follow these instructions](https://developer.apple.com/documentation/swift_packages/adding_package_dependencies_to_your_app). If you want to learn more about Swift Package Manager, [this page goes into detail](https://swift.org/package-manager/).

> If you're a framework author and want to use SwiftCurrent as a dependency, update your `Package.swift` file:

Reviewer: ```suggestion
If you're a framework author and want to use SwiftCurrent as a dependency, update your `Package.swift` file.
```

the colon leading into a header seemed weird to me.

## Get the package

Add the following line to the package dependencies in `Package.swift`:

```swift

> .package(url: "https://github.com/wwt/SwiftCurrent.git", .upToNextMajor(from: "3.0.0")),

Reviewer: ```suggestion
.package(url: "https://github.com/wwt/SwiftCurrent.git", .upToNextMajor(from: "4.0.0")),
```
The packages down below do not exist prior to version 4.

```

## Get the correct product

Add one one of the following products to your target dependencies.

#### **If you want to use Workflow with UIKit**

```swift
.product(name: "Workflow", package: "Workflow"),
.product(name: "WorkflowUIKit", package: "Workflow")
```

> #### **You'll import Workflow for UIKit as such**

Reviewer 2: "as such" reads poorly, it's difficult to articulate why. I suggest alternate phrasing.

Reviewer 2: Not going to duplicate this feedback but it should change in all places we used it.

Author: Phrasing updated to "Your import statement in this case will be" in [this commit](https://github.com/wwt/SwiftCurrent/pull/36/commits/37804d0febe7f73f07e54dde38bab04010030d58)

> .product(name: "SwiftCurrent_UIKit", package: "SwiftCurrent_UIKit"")

Reviewer: This line has too many quotes in it, also I think it should be the `SwiftCurrent` package with the product being `SwiftCurrent_UIKit`.
```suggestion
.product(name: "SwiftCurrent_UIKit", package: "SwiftCurrent")
```

```swift

> import SwiftCurrent_UIKit

Reviewer: Since it looks like we are saying that you also need `SwiftCurrent` we should include here with maybe a quick why?  I'm on the fence about this though.

Author: I can see that - did you have some verbiage in mind for the why?

Reviewer: Nope.

```

`WorkflowUIKit` will need to be built on a platform that supports UIKit, such as iOS or macOS with Catalyst.

#### **If you want to use SwiftCurrent without UIKit**

```swift
.product(name: "SwiftCurrent", package: "SwiftCurrent"),
```
#### **Your import statement in this case will be**

```swift
import SwiftCurrent
```

You will need to build your own [Orchestration Responders](https://gitcdn.link/cdn/wwt/SwiftCurrent/faf9273f154954848bf6b6d5c592a7f0740ef53a/docs/Protocols/OrchestrationResponder.html) for your domains.

# CocoaPods

Set up [CocoaPods](https://cocoapods.org/) for your project, then include SwiftCurrent in your dependencies by adding one of the following lines to your `Podfile`:

#### **If you want to use SwiftCurrent with UIKit**

```ruby
pod 'SwiftCurrent/UIKit'
```

#### **If you want to use SwiftCurrent without UIKit**

```ruby
pod 'SwiftCurrent/Core'
```

#### **In both of these cases your import statement will be**

```swift
import SwiftCurrent
```

> You will need to build your own [Orchestration Responders](https://gitcdn.link/cdn/wwt/SwiftCurrent/faf9273f154954848bf6b6d5c592a7f0740ef53a/docs/Protocols/OrchestrationResponder.html) for your domains.

Reviewer: This is linked to `SwiftCurrent/Core` and needs to be colocated with it or have a callback to if you used Core.

Author: Made some changes to this verbiage

If you want more information about installing Swift packages with Xcode, [follow these instructions](https://developer.apple.com/documentation/swift_packages/adding_package_dependencies_to_your_app). If you want to learn more about Swift Package Manager, [this page goes into detail](https://swift.org/package-manager/).

> To use Swift Package Manager in Xcode, click `File` -> `Swift Packages` -> `Add Package Dependency` and enter the Workflow repo's URL.

Reviewer: I'm interested in the need of these lines in comparison to line 3 and what line 3 failed to provide that these lines are providing.

Author: For me, I felt that explicitly stating how to get the dependency in is worthwhile even if redundant alongside line 3. I could see still having this sentence from line 3:

"If you want to learn more about Swift Package Manager, [this page goes into detail](https://swift.org/package-manager/)."

But I feel explicitly giving direction here is more easily digestible than linking out as the first sentence in line 3 does. As I'm writing this out, I'm thinking I probably could've combined those to be:

"To use Swift Package Manager in Xcode, click `File` -> `Swift Packages` -> `Add Package Dependency` and enter the Workflow repo's URL. If you want to learn more about Swift Package Manager, [this page goes into detail](https://swift.org/package-manager/)."

Reviewer 2: I'll caution against instructions that rely on the UI. Are you going to update this if Xcode 14 changes the structure? Will you then have to have Xcode < 14 instructions and Xcode 14 instructions? Easier to link out to an Apple thing methinks.

Author: Updated to remove the new lines I had previously added, with a small change to the existing verbiage of line 3.

## Get the correct product

> ### Add one one of the following products to your target dependencies.

Author: No, not sure why - it looked fine without. Will undo.

#### If you want to use Workflow with UIKit

```swift
.product(name: "WorkflowUIKit", package: "Workflow")
```

#### You'll import Workflow for UIKit as such

```swift
import WorkflowUIKit
```

`WorkflowUIKit` will need to be built on a platform that supports UIKit, such as iOS or macOS with Catalyst.

#### If you want to use Workflow without UIKit

```swift
.product(name: "Workflow", package: "Workflow"),
```

#### You'll import Workflow without UIKit as such

```swift
import Workflow
```

You will need to build your own [Orchestration Responders](https://gitcdn.link/cdn/wwt/Workflow/faf9273f154954848bf6b6d5c592a7f0740ef53a/docs/Protocols/OrchestrationResponder.html) for your domains.

> # CocoaPods

Author: I think resolved in [commit 6bba9b1](https://github.com/wwt/Workflow/pull/36/commits/6bba9b1bc32178bcec0be3717e179a10d63f9eb7), which I forgot to push earlier.

## Get the correct product

Add one one of the following products to your target dependencies.

> #### __If you want to use Workflow with UIKit__

Reviewer: Are we using __ or ** in other files?  Is this something we should have a preference on?

Author: We are using ** elsewhere and that's what GitHub does by default as well, so I will replace.

```swift
import SwiftCurrent
```

> `Note:` When using SwiftCurrent for a custom domain, you will need to build the associated [Orchestration Responders](https://gitcdn.link/cdn/wwt/SwiftCurrent/faf9273f154954848bf6b6d5c592a7f0740ef53a/docs/Protocols/OrchestrationResponder.html).

Reviewer: `you will need to build the associated [Orchestration Responders]` for some reason made me think that there would be an Orchestration Responder I needed to get and then build as well, versus me having to create it.  Perhaps this would read better?

```suggestion
`Note:` When using SwiftCurrent for a custom domain, you will need to build out the associated [Orchestration Responders](https://wwt.github.io/SwiftCurrent/Protocols/OrchestrationResponder.html).
```
I like this one better, thanks @wiemerm !

pod 'DynamicWorkflow/Core'
```

#### In both of these cases you'll import Workflow as such

```swift
import Workflow

> ```

Author: Is this the truth? I took a look at the podspec and I believe it is, but wanted another check

pod 'SwiftCurrent/Core'
```

> `Note:` When using SwiftCurrent for a custom domain, you will need to build the associated [Orchestration Responders](https://gitcdn.link/cdn/wwt/SwiftCurrent/faf9273f154954848bf6b6d5c592a7f0740ef53a/docs/Protocols/OrchestrationResponder.html).

Reviewer: same as above:
```suggestion
`Note:` When using SwiftCurrent for a custom domain, you will need to build out the associated [Orchestration Responders](https://wwt.github.io/SwiftCurrent/Protocols/OrchestrationResponder.html).
```
